logagg_collector/collector.py

```diff
@@ -0,0 +1,926 @@
+        Sample url: 'http://localhost:6600/collector/v1/start'
+        '''
+        self.collector.collect()
+
+        return dict(self.collector.state)
+
+    def stop(self) -> dict:
+        '''
+        Sample url: 'http://localhost:6600/collector/v1/stop'
+        '''
+        sys.exit(0)
+
+
+    def add_file(self, fpath:str, formatter:str) -> list:
+        '''
+        Sample url: 'http://localhost:6600/collector/v1/add_file?fpath="/var/log/serverstats.log"&formatter="logagg_collector.formatters.docker_file_log_driver"'
+        '''
+        f = {"fpath":fpath, "formatter":formatter}
+
+        # Add new file to logaggfs trackfiles.txt
+        self.collector.add_to_logaggfs_trackfile(f['fpath'])
+
+        # Add new file to state
+        state = self.collector.state['fpaths']
+        # If fpath already present; remove and update
+        if f in state: state.remove(f)
+        state.append(f)
+        self.collector.state['fpaths'] = self.collector.fpaths = state
+        self.collector.state.flush()
+
+        return self.collector.state['fpaths']
+
+
+    def get_files(self) -> list:
+        '''
+        List of file patterns to be tracked by collector
+        '''
+        return self.collector.state['fpaths']
+
+
+    def _get_nsq(self):
+        '''
+        Returns NSQ details
+        '''
+        return dict(nsqd_http_address = self.collector.nsq_sender_logs.nsqd_http_address,
+                topic_name = self.collector.nsq_sender_logs.topic_name)
+
+
+    def set_nsq(self, nsqd_http_address:str, topic_name:str) -> dict:
+        '''
+        Takes details of NSQ to which formatted logs are to be sent
+        '''
+        nsq_sender_logs = NSQSender(nsqd_http_address, topic_name, self.log)
+        self.collector.nsq_sender_logs = nsq_sender_logs
+
+        return self._get_nsq()
+
+
+    def get_nsq(self) -> dict:
+        '''
+        Returns NSQ details
+        '''
+        return self._get_nsq()
+
+
+    def get_active_log_collectors(self) -> list:
+
+        # return the files on which active threads are running to collect logs
+        collectors = self.collector.log_reader_threads
+        c = [t[0] for t in collectors if collectors[t].isAlive()]
+        return c
+
+
+    def remove_file(self, fpath:str) -> dict:
+        '''
+        Sample url: 'http://localhost:6600/collector/v1/add_file?fpath="/var/log/serverstats.log"'
+        '''
+        #FIXME: stops the program after removing
+
+        # Remove fpath from logaggfs trackfile.txt
+        self.collector.remove_from_logaggfs_trackfile(fpath)
+
+        # Remove fpath from state
+        s = list()
+        for f in self.collector.state['fpaths']:
+            if f['fpath'] == fpath: pass
+            else: s.append(f)
+
+        self.collector.state['fpaths'] = s
+        self.collector.state.flush()
+        self.log.info('exiting', fpaths=self.collector.state['fpaths'])
+        self.log.info('restart_for_changes_to_take_effect')
+        sys.exit(0)
```

Comment:
If we are exiting at this the control of your program will never go to return statement. please do check this and do need ful.

Reply:
I think you missed this.